nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/RestLookupService.java

```diff
@@ -0,0 +1,390 @@
+                    return response.request().newBuilder()
+                            .header("Proxy-Authorization", credential)
+                            .build();
+                });
+            }
+        }
+    }
+
+    @Override
+    public Optional<Record> lookup(Map<String, Object> coordinates) throws LookupFailureException {
+        final String endpoint = (String)coordinates.get(ENDPOINT_KEY);
+        final String mimeType = (String)coordinates.get(MIME_TYPE_KEY);
+        final String method   = (String)coordinates.get(METHOD_KEY);
+        final String body     = (String)coordinates.get(BODY_KEY);
+
+        if (StringUtils.isBlank(body) && (method.equals("post") || method.equals("put"))) {
+            throw new LookupFailureException(
+                String.format("Used HTTP verb %s without specifying the %s key to provide a payload.", method, BODY_KEY)
+            );
+        }
+
+        Request request = buildRequest(mimeType, method, body, endpoint);
+        try {
+            Response response = executeRequest(request);
+            InputStream is = response.body().byteStream();
+
+            Record record = handleResponse(is, coordinates);
+
+            return record != null
+                    ? Optional.of(record)
+                    : Optional.empty();
+        } catch (Exception e) {
+            getLogger().error("Could not execute lookup.", e);
+            throw new LookupFailureException(e);
+        }
+    }
+
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        if (propertyDescriptorName.startsWith("header")) {
+            String header = propertyDescriptorName.substring(propertyDescriptorName.indexOf(".") + 1);
+            return new PropertyDescriptor.Builder()
+                .name(propertyDescriptorName)
+                .displayName(header)
+                .addValidator(Validator.VALID)
+                .dynamic(true)
+                .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+                .build();
+        }
+
+        return null;
+    }
+
+    protected Response executeRequest(Request request) throws IOException {
+        return client.newCall(request).execute();
+    }
+
+    private Record handleResponse(InputStream is, Map<String, Object> coordinates) throws SchemaNotFoundException, MalformedRecordException, IOException {
+        Map<String, String> variables = coordinates.entrySet().stream()
+            .collect(Collectors.toMap(
+                e -> e.getKey(),
+                e -> e.getValue().toString()
+            ));
+        try (RecordReader reader = readerFactory.createRecordReader(variables, is, getLogger())) {
+
+            Record record = reader.nextRecord();
+
+            if (recordPath != null) {
+                Optional<FieldValue> fv = recordPath.evaluate(record).getSelectedFields().findFirst();
+                if (fv.isPresent()) {
```

Comment:
If this evaluates to `false` (i.e., the RecordPath doesn't match anything) then we return the entire Record. I don't think that's what we want here? Should we return `null` in that case? We should probably document the behavior, whatever it is.

Reply:
Done (minus docs).